packages/orbit-components/src/InputSelect/index.tsx

```diff
@@ -0,0 +1,569 @@
+      const { grouped, ungrouped } = options.reduce(
+        (acc, option) => {
+          const { group } = option;
+          if (!group) {
+            acc.ungrouped.push(option);
+            return acc;
+          }
+
+          // Check the first element of each array to see if it's from the same group.
+          // If it is, add the option to that array.
+          // If not, create a new array with the group as the first element and add the option to that array.
+          const found = acc.grouped.find(arr => arr[0] && arr[0].group === group);
+          if (found) {
+            found.push(option);
+          } else {
+            acc.grouped.push([option]);
+          }
+
+          return acc;
+        },
+        { grouped: [] as Option[][], ungrouped: [] as Option[] },
+      );
+
+      if (previouslySelected) {
+        grouped.unshift([previouslySelected]);
+      }
+
+      const flattenedGroups = grouped.reduce((acc, group) => [...acc, ...group], [] as Option[]);
+      const flattened = showAll
+        ? [...flattenedGroups, ...options]
+        : [...flattenedGroups, ...ungrouped];
+
+      return {
+        groups: grouped,
+        all: options,
+        ungrouped,
+        flattened,
+      };
+    }, [options, previouslySelected, showAll]);
+
+    const [results, setResults] = React.useState<{
+      groups: Option[][];
+      all: Option[];
+      flattened: Option[];
+    }>(groupedOptions);
+
+    const handleFocus = (ev: React.SyntheticEvent<HTMLInputElement>) => {
+      if (onFocus) onFocus(ev);
+      setIsOpened(true);
+      setResults(results || groupedOptions);
+    };
+
+    const handleBlur = (ev: React.SyntheticEvent<HTMLInputElement>) => {
+      if (onBlur) onBlur(ev);
+    };
+
+    const handleInputChange = (ev: React.SyntheticEvent<HTMLInputElement>) => {
+      const { value } = ev.currentTarget;
+      if (onChange) onChange(ev);
+
+      if (value.length === 0) {
+        setResults(groupedOptions);
+      } else {
+        const filtered = options.filter(({ title }) => {
+          return title.toLowerCase().includes(value.toLowerCase());
+        });
+        setResults({
+          groups: [],
+          all: filtered,
+          flattened: filtered,
+        });
+      }
+
+      setInputValue(value);
+      setActiveIdx(0);
+    };
+
+    const handleDropdownKey = (ev: React.KeyboardEvent<HTMLInputElement>) => {
+      if (isOpened && ev.keyCode === KEY_CODE.ESC) setIsOpened(false);
+      if (!isOpened && ev.keyCode === KEY_CODE.ENTER) setIsOpened(true);
+
+      if (isOpened && (ev.keyCode === KEY_CODE.ENTER || ev.keyCode === KEY_CODE.SPACE)) {
+        ev.preventDefault();
+
+        if (results.all.length !== 0) {
+          setSelectedOption(results.flattened[activeIdx]);
+          setIsOpened(false);
+          setInputValue(results.flattened[activeIdx].title);
+        }
+      }
+
+      if (ev.keyCode === KEY_CODE.ARROW_DOWN) {
+        if (!isOpened) {
```

Comment:
I am not sure I understand your concern. I don't see the dropdown opening without focus.

My thought process to add this was the following use case: a user, using only the keyboard, focuses on the input. It opens automatically. Uses the keys to chose an option. Confirms with ENTER. But then wants to change their selection. By pressing the arrow down key, nothing happens. And I would expect the list of options to open and be able to navigate through them.

In fact, if I remove this, I just noticed that the cycle between options continues, internally. And if the user opens the dropdown again (with the ENTER key) it will be on a completely different option. And this is also happening on the arrow up. So regardless of opening the dropdown or not with the arrows, I will need to add a verification to see if the dropdown is open.

Now, decision: should the dropdown open when the input is focused and one presses the arrows? (I don't see it happening when the input is not focused 🤔)

Reply:
Aha, ok, thanks for the explanation
from the code, it just looked to me that you add that in order to just open that 👍🏻 

> _Now, decision: should the dropdown open when the input is focused and one presses the arrows? (I don't see it happening when the input is not focused 🤔)_

it's already opening on focus :) 

Sometimes for a11y people add a combination of keys, like alt+arrow-down I mentioned above, in order to open the dropdown without focusing input. You can take a look [here](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/)

Reply:
I was asking if it should happen when you press an arrow and it's already focused. But I see we agree it should 😄 

I think we can consider combination of keys for a future iteration. As I feel it would also be useful for other Orbit components, so we could unify